### PR TITLE
feat: add completion if url and add version select

### PR DESCRIPTION
**Pull request by hblitza**
This checks url input for the search params  
- request
- version
- service  
and adds them if needed.  
+ adds version select box
+ some layout adaptions (spaces)

### CODE REVIEW

File: src/components/AddLayerModal/index.tsx [hunk 1]
```diff
@@ -81,6 +84,28 @@ export const AddLayerModal: React.FC<AddLayerModalProps> = ({
...
+  /**
+   * Checks and completes a WMS GetCapabilities URL.
+   * The params request, service and version will be completed.
+   * @param url The input url
+   * @returns The completed URL
+   */
+  const completeUrl = (url: string) => {
+    let completeUrl = new URL(url);
+
+    if (!completeUrl.searchParams.has('request')) {
```
[the remark this answers is not in the document]
Comment (hblitza) [reply src/components/AddLayerModal/index.tsx:96]: right

File: src/components/AddLayerModal/index.tsx [hunk 2]
```diff
@@ -81,6 +84,28 @@ export const AddLayerModal: React.FC<AddLayerModalProps> = ({
     }
   };
 
+  /**
+   * Checks and completes a WMS GetCapabilities URL.
+   * The params request, service and version will be completed.
+   * @param url The input url
+   * @returns The completed URL
+   */
+  const completeUrl = (url: string) => {
```
Comment (dnlkoch) [inline src/components/AddLayerModal/index.tsx:93]: We might want to reuse [this method](https://github.com/terrestris/base-util/blob/master/src/UrlUtil/UrlUtil.ts#L129) from the base-util instead.

**Review by dnlkoch — COMMENTED** [review]
Very nice @hblitza!

Apart from the suggestion about the base-util there are some linter warnings left we might want to have a look at.

File: src/components/AddLayerModal/index.tsx [hunk 3]
```diff
@@ -169,33 +186,54 @@ export const AddLayerModal: React.FC<AddLayerModalProps> = ({
...
-      />
-      <Table
-        loading={loading}
-        columns={[
-          {
-            title: t('AddLayerModal.columnTitle'),
-            render: (text: any, record: any) => {
-              return record.get('title');
-            }
+      <Space direction="vertical"
```
Comment (dnlkoch) [inline src/components/AddLayerModal/index.tsx:189]:
```suggestion
      <Space 
        direction="vertical"
```

File: src/components/AddLayerModal/index.tsx [hunk 4]
```diff
@@ -169,33 +186,54 @@ export const AddLayerModal: React.FC<AddLayerModalProps> = ({
...
+        style={{ display: 'flex' }}
```
Comment (dnlkoch) [inline src/components/AddLayerModal/index.tsx:190]: Can we move this to the style file? And do we even need the `Space` element?

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**